File: android/src/main/java/com/twiliovoicereactnative/ExpoApplicationLifecycleListener.java
package com.twiliovoicereactnative;

import android.app.Application;
import expo.modules.core.interfaces.ApplicationLifecycleListener;

// Expo Modules API用: アプリケーションのライフサイクルイベントをTwilio Voice SDKに伝えるリスナー
public class ExpoApplicationLifecycleListener implements ApplicationLifecycleListener {
    VoiceApplicationProxy voiceApplicationProxy;

    @Override
    public void onCreate(Application application) {
        this.voiceApplicationProxy = new VoiceApplicationProxy(application);
        this.voiceApplicationProxy.onCreate();
    }
}

File: android/src/main/java/com/twiliovoicereactnative/ExpoApplicationLifecycleListener.kt
package com.twiliovoicereactnative

import android.app.Application
import expo.modules.core.interfaces.ApplicationLifecycleListener

class ExpoApplicationLifecycleListener : ApplicationLifecycleListener{
    private var voiceApplicationProxy: VoiceApplicationProxy? = null

    override fun onCreate(application: Application?) {
        if (application != null) {
            voiceApplicationProxy = VoiceApplicationProxy(application)
        }

        voiceApplicationProxy?.onCreate()

        super.onCreate(application)
    }
}
